### DISCUSSION TITLE
Remember Me for Identity Providers

### DISCUSSION THREAD

**Discussion by sirishkumar** [keycloak/keycloak#15513] [Q&A]
I am trying to figure out how to implement remember me functionality for identity providers. After doing a bit of debugging with keycloak code, my understanding is we need to somehow convert the following session cookies to persistent cookies which are created by AuthenticationManager
- KEYCLOAK_IDENTITY_LEGACY
- KEYCLOAK_IDENTITY
- KEYCLOAK_SESSION_LEGACY
- KEYCLOAK_SESSION

I did not find an easy way for overriding the creation of these cookies using a custom authenticator, but what worked for me is running a custom authenticator as part of Post login flow in our custom identity provider as done by username and password authenticator. Code reference [here](https://github.com/keycloak/keycloak/blob/e396d0daa1fed0cb82cf761fe3b461ba24118a10/services/src/main/java/org/keycloak/authentication/authenticators/browser/AbstractUsernameFormAuthenticator.java#L205-L218)

```
Details = Java.type("org.keycloak.events.Details");

function authenticate(context) {
    context.getAuthenticationSession().setAuthNote(Details.REMEMBER_ME, "true");
    context.success();
}
```

Similar code in Java

```
import org.keycloak.authentication.AuthenticationFlowContext;
import org.keycloak.authentication.Authenticator;
import org.keycloak.events.Details;

public class IdPRememberMeAuthenticator implements Authenticator {
    @Override
    public void authenticate(AuthenticationFlowContext context) {
        context.getAuthenticationSession().setAuthNote(Details.REMEMBER_ME, "true");
        context.success();
    }
}
```

But not sure this is the right way of doing this and the implications of it. Can someone comment?

I also mentioned this in a post on Discourse, [link](https://keycloak.discourse.group/t/guidance-for-enabling-remember-me-functionality-for-social-providers/7188/4)

**Comment by mandree95** [comment]
Is it possible to add it to the idp settings? At least I didn't find any configuration options. Is this on the roadmap?
